analyses/malawi/dryspells_trigger/07_ecmwf_monthly_skill_general.md

```diff
@@ -41,164 +37,372 @@ reload(processing)
+#                                           da_forecast,
+#                                           normalization=n
+#                                          ).rename(columns={"crps":f"{n_str}crps"})
+#     df_crps=pd.concat([df_crps,df_crps_all],axis=1)
+#     for thresh in thresh_list:
+#         df_crps_th=processing.get_crps_ecmwf(da_obs,
+#                                              da_forecast,
+#                                              normalization=n,
+#                                              thresh=thresh
+#                                             ).rename(columns={"crps":f"{n_str}crps_{thresh}"})
+#         df_crps=pd.concat([df_crps,df_crps_th],axis=1)
+#     for m in sel_m:
+#         month_str="".join([calendar.month_abbr[i].lower() for i in m])
+#         da_obs_m = da_obs.where(da_obs.time.dt.month.isin(m), drop=True)
+#         da_forecast_m = da_forecast.where(da_forecast.time.dt.month.isin(m), drop=True)
+#         df_crps_m=processing.get_crps_ecmwf(da_obs_m,
+#                                             da_forecast_m,
+#                                             normalization = n,
+#                                            ).rename(columns={"crps":f"{n_str}crps_{month_str}"})
+#         df_crps=pd.concat([df_crps,df_crps_m],axis=1)
+#         for thresh in thresh_list:
+#             df_crps_m=processing.get_crps_ecmwf(da_obs_m,
+#                                                 da_forecast_m,
+#                                                 thresh=thresh,
+#                                                 normalization = n,
+#                                                ).rename(columns={"crps":f"{n_str}crps_{month_str}_{thresh}"})
+#             df_crps=pd.concat([df_crps,df_crps_m],axis=1)
+# df_crps.to_csv(crps_path)
 ```
 
 ```python
-da_obs=rioxarray.open_rasterio(chirps_monthly_mwi_path,masked=True)
-#only select the years for which we also identified dry spells
-da_obs=da_obs.sel(time=da_obs.time.dt.year.isin(range(2000,2021)))
+df_crps=pd.read_csv(crps_path)
 ```
 
+From the plots below we can see that
+- The CRPS is pretty high, resulting in a 30% error across all values
+- The CRPS is relatively steady across leadtimes. We can see a dip at 1 month leadtime, and a smaller dip at 4 months leadtime
+- The CRPS differs across months and thresholds
+- For the months around the rainy season (Nov-Apr), we can see that the CRPS is relatively high, but the normalized CRPS is lower. This is largely due to the fact that the precipitation during the rainy season is higher. 
+    - While lower during Nov-Apr, and specifically JanFeb, the error is still 25% which is quite large. 
+- When selecting on the precipitation threshold, we can see that the higher the threshold the lower the normalized CRPS. 
+    - This is likely because there are many very low values in the data, which causes the normalized CRPS to get large quickly even though the absolute difference might be small
+
```

Comment:
It's weird that the CRPS doesn't get worse with lead time. Maybe that's a sign that it's not an appropriate metric here

Reply:
I agree it is really weird. At the same time, also when looking at other metrics it doesn't get worse. E.g. for the trigger design the performance of 3 months leadtime was better than that of 2 months. I think this is something we should ask ECMWF.